### PR TITLE
Feature/analysis queue

### PR DESCRIPTION
**Pull request by evenodd**

### CODE REVIEW

File: db/models/jobSchema.js [hunk 1]
```diff
@@ -7,7 +8,15 @@ const JobSchema = mongoose.Schema({
         type: Boolean,
         default: false
     },
-    images: [Buffer]
+    images: [Buffer],
```
Comment (patmizi) [inline db/models/jobSchema.js:11]: What is the purpose of this variable when we have two image fields in the `JobAnalysisSchema`

File: routes/job.js [hunk 2]
```diff
@@ -28,4 +32,72 @@ router.put('/', Utils.asyncRoute(async function (req, res, next) {
...
+router.post('/analysis/:jobId', Utils.asyncRoute(async function (req, res, next) {
+    const {jobId} = req.params;
+    const {image, modelPrediction, analysisQueueId} = req.body
+    const connector = new DbConnector();
+    const timestamp = new Date().getTime();
+    const jobAnalysis = new JobAnalysis({
+        jobId: jobId,
+        timestamp: timestamp,
+        modelPrediction: modelPrediction,
+        image: image,
```
Comment (patmizi) [inline routes/job.js:61]: JobAnalysis is missing the `imageThumbnail` field. You can probably perform the image transforms on the `image` within the object constructor.

File: db/models/jobSchema.js (hunk 1 above)
Comment (evenodd) [reply db/models/jobSchema.js:11]: Yea this is unused. I can clean this up

File: routes/job.js [hunk 3]
```diff
@@ -28,4 +32,72 @@ router.put('/', Utils.asyncRoute(async function (req, res, next) {
...
+        job.analysis.unshift(jobAnalysis);
+
+        siteModel.rooms[0]['lastAnalysedTimestamp'] = timestamp;
+        job.lastAnalysisId = analysisQueueId;
+        job.completed = modelPrediction >= 0.85;
+
+        await siteModel.save();
+
+        const analysisQueue = await AnalysisQueue.findById(analysisQueueId).exec();
+        const siteModel = await SiteModel.findOne(
```
Comment (patmizi) [inline routes/job.js:78]: Duplicate declaration of variable

File: routes/job.js [hunk 4]
```diff
@@ -28,4 +32,72 @@ router.put('/', Utils.asyncRoute(async function (req, res, next) {
...
+    const jobAnalysis = new JobAnalysis({
+        jobId: jobId,
+        timestamp: timestamp,
+        modelPrediction: modelPrediction,
+        image: image,
+    });
+    try {
+        connector.connect();
+
+        const siteModel = await SiteModel.findOne({'rooms.jobs._id': mongoose.Types.ObjectId(jobId)}, {'rooms.jobs.$': true});
```
Comment (patmizi) [inline routes/job.js:66]: Duplicate declaration of variable

File: routes/job.js (hunk 3 above)
Comment (evenodd) [reply routes/job.js:78]: Yea there is a larger bug in this endpoint im looking at. The analysisQueue will be set as completed even if only one job is analysed.  Ill push the variable name fix for now

**Pull request merged**